rust/chains/hyperlane-cosmos/src/providers/rpc.rs

```diff
@@ -143,15 +111,14 @@ impl WasmIndexer for CosmosWasmIndexer {
         debug!("Query: {:?}", query.to_string());
 
         let tx_search_result = client
-            .tx_search(query.clone(), false, 1, 30, Order::Ascending)
+            .tx_search(query.clone(), false, 1, PAGINATION_LIMIT, Order::Ascending)
             .await?;
 
         let total_count = tx_search_result.total_count;
-        let last_page = total_count / 30 + (total_count % 30 != 0) as u32;
+        let last_page = total_count / PAGINATION_LIMIT as u32
```

Comment:
Just want to double check that it's ok to use `100` as the pagination limit here, given that @hashableric suggested 100 might cause payloads to be too big: https://github.com/many-things/hyperlane-monorepo/pull/12#discussion_r1371538870

Reply:
I haven't confirmed it exactly. However, when 30 transactions are queried, the payload is about 1.8MB. According to this calculation, when querying 100 transactions, you will have a payload of approximately 6MB.

Most gRPC servers have a payload limit of 4MB, which can cause errors depending on which chain node you are querying. @daniel-savu

Reply:
OMG it isnt GRPC yes Its quite enough for just http payload @daniel-savu